src/amo/components/Categories.js

```diff
@@ -0,0 +1,71 @@
+import React, { PropTypes } from 'react';
+import { compose } from 'redux';
+
+import Link from 'amo/components/Link';
+import translate from 'core/i18n/translate';
+
+import './Categories.scss';
+
+
+export function filterAndSortCategories(categories, addonType, clientApp) {
+  return categories.filter((category) => {
+    return category.type === addonType && category.application === clientApp;
+  }).sort((a, b) => {
+    return a.name > b.name;
+  });
+}
+
+export class CategoriesBase extends React.Component {
+  static propTypes = {
+    addonType: PropTypes.string.isRequired,
+    categories: PropTypes.arrayOf(PropTypes.object),
+    clientApp: PropTypes.string.isRequired,
+    error: PropTypes.bool,
+    loading: PropTypes.bool.isRequired,
+    i18n: PropTypes.object.isRequired,
+  }
+
+  render() {
+    const {
+      addonType, categories, clientApp, error, loading, i18n,
+    } = this.props;
+
+    if (loading && !categories.length) {
+      return <div>{i18n.gettext('Loading...')}</div>;
```

Comment:
Maybe these `<div>`s should have the `Category` class, and potentially a `<p>` inside of them. That way any general layout stuff is the same.